src/Tools/IiifHelper.php

```diff
@@ -28,31 +29,32 @@ class IiifHelper {
   protected $httpClient;
 
 
-
-  public function __construct($remoteUrl) {
+  public function __construct($publicUrl, $internalUrl) {
     //@TODO verify that remoteURL is a valid, actual IIIF info json URL.
-    $this->remoteInfoJsonURL = $remoteUrl;
     $this->httpClient = \Drupal::httpClient();
-    $this->remoteInfoJson = $this->getRemoteIIIFImageInfo($remoteUrl);
+    $this->publicUrl = $publicUrl;
+    $this->internalUrl = $internalUrl;
   }
 
-  public function getRemoteIIIFImageInfo($remoteUrl) {
+  public function getRemoteInfoJson($id) {
```

Comment:
some of the functions in this class require type hinting for the arguments and PHP_DOCS. Please complete that part. thanks! What local PHP are you running? I check normally against 7.1 and 7.2, 7.3 lately too but not mandatory